tests/test_bigbio.py

```diff
@@ -536,18 +606,97 @@ def test_schema(self, schema: str):
         else:
             features = _SCHEMA_TO_FEATURES[schema]
 
-        split_to_feature_counts = self.get_feature_statistics(features=features, schema=schema)
-        for split_name, split in self.datasets_bigbio[schema].items():
-            self.assertEqual(split.info.features, features)
-            for non_empty_feature in non_empty_features:
-                if split_to_feature_counts[split_name][non_empty_feature] == 0:
-                    raise AssertionError(f"Required key '{non_empty_feature}' does not have any instances")
+        split_to_feature_counts = self.get_feature_statistics(
+            features=features, schema=schema
+        )
+
+        # Omit a particular key in a split
+        if len(self.BYPASS_SPLIT_AND_KEY) > 0:
```

Comment:
this seems like a lot of extra code ... is it possible to frame this as an early return condition in the original loop? 
```
for split_name, split in self.datasets_bigbio[schema].items():
    if split_name in self.BYPASS_SPLITS:
        continue  ? 
    self.assertEqual(split.info.features, features)
    for non_empty_feature in non_empty_features:
        if split_to_feature_counts[split_name][non_empty_feature] == 0:
            raise AssertionError(f"Required key '{non_empty_feature}' does not have any instances")
```

Reply:
`BYPASS_SPLIT_AND_KEY` is a specific condition

In the above implementation, you can either bypass a data split (i.e. ignore ALL testing for "train"), bypass a key (i.e. "ignore events"). This applies uniformly to all cases.

In certain cases (i.e. many of the bionlp datasets), the test set is the only dataset to have only 1 key missing (for example, the test set is missing "events" in an event extraction task). For this case, it's wasteful and possibly dangerous to waive testing on the ENTIRE split, when the input examples (like entities) may need to be checked. The logic splits because in certain cases you need a *specific* combination of (split name, key to ignore), hence the overhead here.

If you have a suggestion that can streamline it, please feel free. This isn't written particularly efficiently but it works on the cases I've tested to ensure both the new args, and old behavior is maintained.

Reply:
I might be missing something (and if this implementation is totally off, we can ignore it) but I was thinking all 3 conditions could be incorporated with early returns .... 
```
for split_name, split in self.datasets_bigbio[schema].items():

    # skip entire split
    if split_name in self.BYPASS_SPLITS:
        logger.info(f"skipping {split_name}")
        continue  

    self.assertEqual(split.info.features, features)

    for non_empty_feature in non_empty_features:
        # skip specific split,key combo 
        split_key = [split_name, non_empty_feature]
        if split_key in self.BYPASS_SPLIT_KEY:
            logger.info(f"skipping {split_key}")

        # skip key in every split
        if non_empty_features in self.BYPASS_KEYS:
            logger.info(f"skipping {non_empty_feature}")   

        if split_to_feature_counts[split_name][non_empty_feature] == 0:
            raise AssertionError(f"Required key '{non_empty_feature}' does not have any instances")
```

Reply:
ps, the above was just me trying to do some code re-factor ... if we want to just merge these updates as is and cleanup after the hackathon, I'm OK with that too!

Reply:
this is good - i dont mind implementing this albeit AT earliest sometime tomorrow. If you can replace the logic feel free to pull the branch or edit directly on the file.

Reply:
code-refactors are good; as mentioned earlier, I wrote this pretty inelegantly to just get it done. Always good to have a second pair of eyes to squash some of the boilerplate.